### PR TITLE
PEAR-661: hide range control if there is no data

### PR DESCRIPTION
**Pull request by craigrbarnes**
## Description
Hides the custom range controls when the facet does not have data. There is a change from the ticket:
If the custom range value does not result in data for that facet, the custom range is shown; otherwise, the user can only clear the filters from the query expression.

Note: the Age facets have the Days/Years Segment rendered in the parent component. A handler is now passed to `RangeInputWithPrefixedRanges` to hide it if there are no values. This should be refactored later.

This also includes undefined value fixes with custom facets.

## Checklist

- [ ] Added proper unit tests
- [x] Left proper TODO messages for any remaining tasks
- [x] Scanned for web accessibility with **aXe**, and mitigated or documented
      flagged issues

### Example of Year of Diagnosis with no data
image

### Example of Year of Diagnosis with values that result in no data:
image

### Age of Diagnosis with no data:
image

### CODE REVIEW

**Comment by wteouchicago** [comment]
functionality looks good. 👍 
any chance we can add a bit of padding to the bottom (below the "No data for this field" text)? Or would https://github.com/NCI-GDC/gdc-frontend-framework/pull/988 take care of it?

**Comment by craigrbarnes** [comment]
> functionality looks good. 👍 any chance we can add a bit of padding to the bottom (below the "No data for this field" text)? Or would #988 take care of it?

I will fix it in this one.

File: packages/portal-proto/src/features/facets/NumericRangeFacet.tsx [hunk 1]
```diff
@@ -546,6 +548,21 @@ const RangeInputWithPrefixedRanges: React.FC<
     setIsGroupExpanded(!isGroupExpanded);
   };
 
+  // informs the parent component if there is data or no data
+  // only used by the DaysOrYears component
+  useEffect(() => {
```
Comment (paribartandhakal) [inline packages/portal-proto/src/features/facets/NumericRangeFacet.tsx]: I think it's better to use `useDeepCompareEffect` given that `filterValues` is an object. Also, I feel like we are updating other `useEffect` -> `useDeepCompareEffect` as we work on things gradually. I am not sure if it has any performance impact.

Comment (paribartandhakal) [reply packages/portal-proto/src/features/facets/NumericRangeFacet.tsx]: Also a reminder to add padding as Wendy suggested. (commenting as i didn't see it in the newer commits)

Comment (craigrbarnes) [reply packages/portal-proto/src/features/facets/NumericRangeFacet.tsx]: Agreed on both. I will do that

Comment (craigrbarnes) [reply packages/portal-proto/src/features/facets/NumericRangeFacet.tsx]: Both suggestions are completed.

**Review by paribartandhakal — APPROVED** [review]

**Pull request merged**